### PR TITLE
Di 22093 where clause with not like cause an exception

### PR DESCRIPTION
**Pull request by dorefaeli-pepperi**

### CODE REVIEW

**Review by ShirMarciano — APPROVED** [review]

**Comment by dorefaeli-pepperi** [comment]
@Chasky-H 
Can you please review the changes and merge?

File: filters/string-filter.ts [hunk 1]
```diff
@@ -97,6 +97,12 @@ export class StringFilter extends Filter {
                 return wildcardQuery(this.apiName, `${this.filterValues[0]}*`);
             case 'EndWith':
                 return wildcardQuery(this.apiName, `*${this.filterValues[0]}`);
+            case 'DoesNotContains':
+                return res.mustNot(wildcardQuery(this.apiName, `*${this.filterValues[0]}*`));
+            case 'DoesNotStartWith':
+                return res.mustNot(wildcardQuery(this.apiName, `${this.filterValues[0]}*`));
+            case 'DoesNotEndWith':
+                return res.mustNot(wildcardQuery(this.apiName, `*${this.filterValues[0]}`));
```
Comment (Chasky-H) [inline filters/string-filter.ts:105]: can you please add support for all functions.

File: json-filter.ts [hunk 2]
```diff
@@ -2,7 +2,15 @@ export type BasicOperations = 'IsEmpty' | 'IsNotEmpty' | 'IsEqual' | 'IsNotEqual
...
 export type NumberOperation = BasicOperations | '=' | '>' | '>=' | '<' | '<=' | '!=' | 'Between';
 
-export type StringOperation = BasicOperations | 'Contains' | 'StartWith' | 'EndWith' | 'IsLoggedInUser';
+export type StringOperation =
+    | BasicOperations
+    | 'Contains'
+    | 'StartWith'
+    | 'EndWith'
+    | 'IsLoggedInUser'
+    | 'DoesNotContains'
```
Comment (Chasky-H) [inline json-filter.ts:11]: DoesNotContain is more proper IMO

File: sql-where-parser.ts [hunk 3]
```diff
@@ -227,6 +243,32 @@ export class SQLWhereParser {
...
+                            // NOT LIKE 'abcd%'
+                            val = val.slice(0, val.length - 1);
+                            res = 'DoesNotStartWith';
+                        } else {
+                            // NOT LIKE 'abcd'
+                            res = 'IsNotEqual';
+                        }
+                        values[0] = val;
+                    }
+                    break;
```
Comment (Chasky-H) [inline sql-where-parser.ts:270]: Is there any way to refactor this code in the LIKE part to reuse here?

**Comment by Chasky-H** [comment]
please add unit tests

File: filters/string-filter.ts (hunk 1 above)
Comment (ShirMarciano) [reply filters/string-filter.ts:105]: What do you means by all functions?

**Comment by Chasky-H** [comment]
sorry, did not mean to close :)

File: filters/string-filter.ts (hunk 1 above)
Comment (Chasky-H) [reply filters/string-filter.ts:105]: string-filter has implementations for each operation SQL clause, filter function, Dynamo function and DSL

Comment (dorefaeli-pepperi) [reply filters/string-filter.ts]: @Chasky-H added to all except DynamoDB, I'm not familiar with the syntax of DynamoDB at all and the implementation of other operations on DynamoDB looks way to complicated to copy and modify for my needs.
How crucial it is to add support for DynamoDB?

**Review by Chasky-H — APPROVED** [review]

**Pull request merged**